x/logic/interpreter/fs/fs.go

```diff
@@ -0,0 +1,99 @@
+package fs
+
+import (
+	goctx "context"
+	"io"
+	"io/fs"
+	"time"
+)
+
+// FileSystem is the custom virtual file system used into the blockchain.
+// It will hold a list of handler that can resolve file URI and return the corresponding binary file.
+type FileSystem struct {
```

Comment:
If the type must obey an interface, and this is the case here, it is better to make an assignment on a blank identifier with a cast on the interface, for type safety:

```
var _ fs.FS = (*FileSystem)(nil)
```